addons/libs/images.lua

```diff
@@ -0,0 +1,344 @@
+    for _, event in ipairs(meta[t].events[event]) do
+        event(t, meta[t].root_settings)
+    end
+end
+
+local apply_settings = function(_, t, settings)
+    settings = settings or meta[t].settings
+    images.pos(t, settings.pos.x, settings.pos.y)
+    images.visible(t, meta[t].status.visible)
+    images.alpha(t, settings.color.alpha)
+    images.color(t, settings.color.red, settings.color.green, settings.color.blue)
+    images.path(t, settings.texture.path)
+    images.fit(t, settings.texture.fit)
+    images.repeat_xy(t, settings.repeatable.x, settings.repeatable.y)
+
+    call_events(t, 'reload')
+end
+
+function images.new(str, settings, root_settings)
+    if type(str) ~= 'string' then
+        str, settings, root_settings = '', str, settings
+    end
+
+    -- Sets the settings table to the provided settings, if not separately provided and the settings are a valid settings table
+    if not _libs.config then
+        root_settings = nil
+    else
+        root_settings =
+            root_settings and class(root_settings) == 'Settings' and
+                root_settings
+            or settings and class(settings) == 'Settings' and
+                settings
+            or
+                nil
+    end
+
+    t = {}
+    local m = {}
+    meta[t] = m
+    m.name = (_addon and _addon.name or 'image') .. '_gensym_' .. tostring(t):sub(8) .. '_%.8X':format(16^8 * math.random()):sub(3)
+    m.settings = settings or {}
+    m.status = m.status or {visible = false, image = {}}
+    m.root_settings = root_settings
+    m.base_str = str
+
+    m.events = {}
+
+    m.keys = {}
+    m.values = {}
+    m.imageorder = {}
+    m.defaults = {}
+    m.formats = {}
+    m.images = {}
+
+    windower.prim.create(m.name)
+
+    amend(m.settings, default_settings)
+    if m.root_settings then
+        config.save(m.root_settings)
+    end
+
+    if _libs.config and m.root_settings and settings then
+        _libs.config.register(m.root_settings, apply_settings, t, settings)
+    else
+        apply_settings(_, t, settings)
+    end
+
+    return setmetatable(t, _meta.Image)
+end
+
+function images.clear(t)
+    local m = meta[t]
+    m.keys = {}
+    m.values = {}
+    m.imageorder = {}
+    m.images = {}
+    m.defaults = {}
+    m.formats = {}
+end
+
+-- Makes the primitive visible
+function images.show(t)
+    windower.prim.set_visibility(meta[t].name, true)
+    meta[t].status.visible = true
+end
+
+-- Makes the primitive invisible
+function images.hide(t)
+    windower.prim.set_visibility(meta[t].name, false)
+    meta[t].status.visible = false
+end
+
+--[[
+    The following methods all either set the respective values or return them, if no arguments to set them are provided.
+]]
+
+function images.pos(t, x, y)
+    local m = meta[t]
+    if not x then
+        return m.settings.Pos.X, m.settings.Pos.Y
+    end
+
+    local settings = windower.get_windower_settings()
+    windower.prim.set_position(m.name, x, y)
+    m.settings.Pos.X = x
```

Comment:
I'm almost certain this won't work, because Windower settings are not the same as addon settings, and they are all *snake_cased*.